### DISCUSSION TITLE
Never true condition

### DISCUSSION THREAD

**Discussion by nobu** [ruby/rubygems#5818] [General]
### The never-true condition
https://github.com/rubygems/rubygems/blob/83297265aa75b608c75e395b4bb06ffa78a9b52d/lib/rubygems/specification.rb#L1098

As you know, `false` is the only instance of `FalseClass`.
If `spec` is `false` the RHS will not be evaluated of course and this expression cannot be true usually.
I don't think gemspec files are using classes overriding `class` method.

### Digging
The RHS of this expression seems originally added to catch in [Gem::BuildCommand#load_gemspec](https://github.com/rubygems/rubygems/commit/688de7f384d67c5ac986e69fdff53c89d637b996#diff-b2e6c07c8b5019a1372062f35cfa4f1c1cb899126d0e19d46543df73c274efd8R321) at 688de7f384d67c5ac986e69fdff53c89d637b996.
This `spec` is the result of `YAML.load` (probably Syck at that time), and it seems to return `false` when the input contains no entities (an empty input, whitespaces or comment only).

Then 2fbc2b66955f7755c10ef8bee21f616e35430b8a added the LHS.
I guess "Raise exception if loaded spec is nil" means [the next block](https://github.com/rubygems/rubygems/commit/2fbc2b66955f7755c10ef8bee21f616e35430b8a#diff-599cc627c77abede6a6e7042edbf4a4a4fb2d196d0b4f232ad738fa3fa9db840R433-R435), as the expression became never-true at this point.
This may be to support Psych.

### Conclusion

I consider this code (and `Gem::EndOfYAMLException` too?) useless, and can be removed.

How do you think about it?

**Comment by deivid-rodriguez** [comment]
Makes sense @nobu, thanks for digging and PR appreciated!

**Comment by deivid-rodriguez** [comment]
@nobu Are you still interested in simplifying this code?
